game/songs.cc

```diff
@@ -82,181 +87,160 @@ void Songs::reload_internal() {
 	doneLoading = true;
 }
 
-#ifdef USE_WEBSERVER
+const std::string SONGS_CACHE_JSON_FILE = "songs.json";
+
 void Songs::LoadCache() {
-	fs::path songsMetaFile = getCacheDir() / "Songs-Metadata.json";
+	const fs::path songsMetaFile = getCacheDir() / SONGS_CACHE_JSON_FILE;
```

Comment:
I'd prefer if you put the JSON read/parse code into its own function, like I did in #545 (plus some comments made by @Baklap4 in there, re: throwing and closing the file, that I just realized I never addressed)

Reply:
The goal is to keep the code as much as possible as it was previously:
- to ease review (code change is difficult to check)
- to stay in line with PR goad (it is just changing the lib used to parse and produce the json songs cache.

About the placement of json.hpp please give me a place where to put it. I think in the game folder it is the best since:
- it is only used for the game
- most discribution do not provide a package for this header
- it enables a nice include : #include "nlohmann/json.hpp"

Reply:
I think @Lord-Kamina has a valid point, but so dos @yoda-jm in terms of the JSON read/parse.
For now let's just plain rewrite it and let's (@Lord-Kamina can you do this?) make an issue to improve/refactor the json capabilities like discussed in #545

> most discribution do not provide a package for this header

Not sure where you did get this info but almost every major system does have it: ubuntu, debian, alpine, arch, freebsd, vcpkg: https://pkgs.org/search/?q=nlohmann
So i think we can depend on the package being installed by the packagemanager. We will need a cmake module for it though.

```
# CMakeLists.txt
find_package(nlohmann_json 3.2.0 REQUIRED)
...
add_library(foo ...)
...
target_link_libraries(foo PRIVATE nlohmann_json::nlohmann_json)
```

Reply:
I think if you add a cmake module and add the dependency to the list of dependencies within game/cmakelists.txt#103
we don't actually have to include the version as it'll probably work with nlohmann3 and nlohmann2

The module should look something like this i think:
FindJson.cmake
```
# - Try to find nlohmann_json
# Once done, this will define
#
#  Json_FOUND - system has nlohmann_json
#  Json_INCLUDE_DIRS - the nlohmann_json include directories
#  Json_LIBRARIES - link these to use nlohmann_json
#
# See documentation on how to write CMake scripts at
# http://www.cmake.org/Wiki/CMake:How_To_Find_Libraries

include(LibFindMacros)
libfind_pkg_detect(Json nlohmann_json FIND_PATH nlohmann/json.hpp FIND_LIBRARY nlohmann_json)
set(Json_VERSION ${Json_PKGCONF_VERSION})
libfind_process(Json)
```

Reply:
outstanding issue at @Tronic LibFindMacros.cmake > https://github.com/Tronic/cmake-modules/issues/5